tests/pc/test_lag_member.py

```diff
@@ -0,0 +1,360 @@
+    """
+    ptfhost.shell("supervisorctl restart ptf_nn_agent")
+
+def remove_ip_from_port(duthost, port, ip=None):
+    """
+    Remove ip addresses from port
+
+    Args:
+        duthost: DUT host object
+        port: port name
+        ip: IP address
+    """
+    ip_addresses = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"].get("INTERFACE", {}).get(port, {})
+    if ip_addresses:
+        for ip in ip_addresses:
+            duthost.shell("config interface ip remove {} {}".format(port, ip))
+    elif ip:
+        duthost.shell("config interface ip remove {} {}".format(port, ip))
+
+def dut_teardown(duthost, dut_lag_map):
+    """
+    Restore dut configuration
+
+    Args:
+        duthost: DUT host object
+        dut_lag_map: imformation about lag in dut
+    """
+    try:
+        for dut_lag_member in dut_lag_map[DUT_LAG_NAME]:
+            duthost.shell("config portchannel member del {} {}".format(DUT_LAG_NAME, dut_lag_member))
+        add_member_to_vlan(duthost, 1000, dut_lag_member, "-u")
+
+        remove_member_from_vlan(duthost, dut_lag_map["vlan"]["id"], DUT_LAG_NAME)
+        duthost.shell("config portchannel del {}".format(DUT_LAG_NAME))
+        remove_member_from_vlan(duthost, dut_lag_map["vlan"]["id"], dut_lag_map[ATTR_PORT_NOT_BEHIND_LAG]["port_name"])
+        remove_ip_from_port(duthost, "Vlan{}".format(dut_lag_map["vlan"]["id"]), dut_lag_map["vlan"]["ip"])
+        add_member_to_vlan(duthost, 1000, dut_lag_map[ATTR_PORT_NOT_BEHIND_LAG]["port_name"], "-u")
+        duthost.shell("config vlan del {}".format(dut_lag_map["vlan"]["id"]))
+    finally:
+        config_reload(duthost)
+
+def ptf_teardown(ptfhost, ptf_lag_map):
+    """
+    Restore ptf configuration
+
+    Args:
+        ptfhost: PTF host object
+        ptf_lag_map: imformation about lag in ptf
+    """
+    ptfhost.shell("ip link set {} nomaster".format(PTF_LAG_NAME))
+
+    for ptf_lag_member in ptf_lag_map[PTF_LAG_NAME]["port_list"]:
+        ptfhost.shell("ip link set {} nomaster".format(ptf_lag_member))
+        ptfhost.shell("ip link set {} up".format(ptf_lag_member))
+        
+    ptfhost.shell("ip link del {}".format(PTF_LAG_NAME))
+    ptfhost.shell("ip addr del {} dev {}".format(ptf_lag_map[ATTR_PORT_NOT_BEHIND_LAG]["ip"], ptf_lag_map[ATTR_PORT_NOT_BEHIND_LAG]["port_name"]))
+    restart_ptf_nn_agent(ptfhost)
+
+def setup_dut_lag(duthost, dut_ports, vlan):
+    """
+    Setup dut lag
+
+    Args:
+        duthost: DUT host object
+        dut_ports: ports need to configure
+        vlan: information about vlan configuration
+
+    Returns:
+        information about dut lag
+    """
+    duthost.shell("config acl remove table EVERFLOW")
+    duthost.shell("config acl remove table EVERFLOWV6")
+    duthost.shell("config portchannel add {}".format(DUT_LAG_NAME))
+
+    lag_port_list = []
+    port_list_idx = 0
+    port_list = list(dut_ports[ATTR_PORT_BEHIND_LAG].values())
+    for _ in range(1, 10000):
```

Comment:
`while len(lag_port_list) < len(dut_ports[ATTR_PORT_BEHIND_LAG])` or `while port_list_idx  < len(dut_ports[ATTR_PORT_BEHIND_LAG])`?

To avoid inf-loop, you can add check before entering the loop such as "if all ports in the vlan are not enough for our testing, then skip this test"

Reply:
Fixed